build test-cases: zig_lib_directory not found in tests/src/Cases.runCases

### Zig Version

0.11.0-dev.6968+84b89d7cf

### Steps to Reproduce and Observed Behavior

```sh
$ mkdir -p build
$ cd build
$ cmake -DCMAKE_BUILD_TYPE=Release -GNinja -DZIG_NO_LIB=ON ..
$ ninja install
```

```sh
build/stage3/bin/zig build test-cases -Dskip-release -Dskip-non-native -fsummary
```

```
x86_64-windows/hello_world_with_updates.0.zig: error: error: FileNotFound
/home/manlio/src/contrib/zig/github.com/perillo/zig/src/introspect.zig:78:5: 0x49ffe9 in findZigLibDirFromSelfExe (check-case)
    return error.FileNotFound;
    ^
/home/manlio/src/contrib/zig/github.com/perillo/zig/src/introspect.zig:51:5: 0x49fb68 in findZigLibDir (check-case)
    return findZigLibDirFromSelfExe(gpa, self_exe_path);
    ^
/home/manlio/src/contrib/zig/github.com/perillo/zig/test/src/Cases.zig:1133:29: 0x49ebdf in runCases (check-case)
    var zig_lib_directory = try introspect.findZigLibDir(self.gpa);
                            ^
/home/manlio/src/contrib/zig/github.com/perillo/zig/test/src/Cases.zig:1122:5: 0x49cf2f in main (check-case)
    return runCases(&ctx, zig_exe_path);
    ^
...
```


### Expected Behavior

No errors for tests run by the check-case executable.

I also noted a unexpected behavior when I install `zig` 0.10.1 via `pacman`:
`build/stage3/bin/zig build test-cases -Dskip-release -Dskip-non-native -fsummary` output is:

```
x86_64-linux/comptime_var.0.zig: error: 
====== expected this output: =========
:4:19: error: store to comptime variable depends on runtime condition
:4:11: note: runtime condition here
␃

======== instead found this: =========
/lib/zig/std/heap.zig:483:39: error: extra capture in for loop
/lib/zig/std/heap.zig:483:39: note: run 'zig fmt' to upgrade your code automatically
...
```
meaning that `test-cases` is using `/bin/zig`.


P.S:
I'm a bit confused with foreign executable handling in the build system.  Recently I noted the `-Dskip-non-native` flag, but `zig build test-cases` will still run windows executables when wine is registered with `binfmt_misc`.

The build system should know when an executable is foreign.

--- a/test/src/Cases.zig
+++ b/test/src/Cases.zig
@@ -1154,7 +1154,7 @@ fn runCases(self: *Cases, zig_exe_path: []const u8) !void {
     progress.terminal = null;
     defer root_node.end();
 
-    var zig_lib_directory = try introspect.findZigLibDir(self.gpa);
+    var zig_lib_directory = try introspect.findZigLibDirFromSelfExe(self.gpa, zig_exe_path);
     defer zig_lib_directory.handle.close();
     defer self.gpa.free(zig_lib_directory.path.?);
 

